protobuf_serialization/reader.nim

```diff
@@ -164,12 +171,28 @@ proc readValueInternal[T: object](stream: InputStream, value: var T, silent: boo
             stream.readFieldPackedInto(fieldVar, header, ProtoType)
           else:
             stream.readFieldInto(fieldVar, header, ProtoType)
-        elif typeof(fieldVar) is ref and defined(ConformanceTest):
+        elif typeof(fieldVar) is ref and defined(npsConformanceTest):
           fieldVar = new typeof(fieldVar)
           stream.readFieldInto(fieldVar[], header, ProtoType)
         else:
           stream.readFieldInto(fieldVar, header, ProtoType)
 
+    # TODO preserve the unknown field
```

Comment:
this should probably be an optional feature - many use cases are fine without the unknown fields (one significantly important use case of unknown field storage is round-tripping and message extension / modification, but if you just want to read a message, it's mostly pointless)